source/routing/isochron.cpp

```diff
@@ -79,10 +79,21 @@ type::GeographicalCoord project_in_direction(const type::GeographicalCoord& cent
     double lat_deg = lat * N_RAD_TO_DEG;
     //If latitude is not in  [-90,90] or longitude is not in [-180,180]
     if (lat_deg < -90 || lat_deg > 90) {
-        lat_deg = (lat_deg / abs(lat_deg)) * 90 + fmod(lat_deg, 90);
+        lat_deg = fmod(lat_deg, 360);
+        if (lat_deg < -90 || lat_deg > 90) {
+            lon_deg =  - lon_deg;
+            if (fabs(lat_deg) > 180) {
+                lat_deg = (std::signbit(lat_deg) ? 1 : -1) * fabs(fmod(lat_deg, 90));
+            } else {
+                lat_deg = (std::signbit(lat_deg) ? - 1 : 1) * (90 - fabs(fmod(lat_deg, 90)));
+            }
+        }
     }
-    if (lon_deg < -180 || lon_deg > 180) {
-        lon_deg = (lon_deg / abs(lon_deg)) * 180 + fmod(lon_deg, 180);
+    if (fmod(lon_deg, 360) < -180 || fmod(lon_deg, 360) > 180) {
+        lon_deg = fmod(lon_deg, 360);
```

Comment:
if 'fmod' keeps the sign intact, you can move this out of 'if' and use it in the test part of the 'if'